contracts/clients/src/BlsSigner.ts

```diff
@@ -0,0 +1,210 @@
+import { ethers, BigNumber, Signer, Bytes } from "ethers";
+import { Deferrable } from "ethers/lib/utils";
+
+import BlsProvider from "./BlsProvider";
+import BlsWalletWrapper from "./BlsWalletWrapper";
+import { ActionData, Bundle, Signature } from "./signer";
+
+export const _constructorGuard = {};
+
+export default class BlsSigner extends Signer {
+  override readonly provider: BlsProvider;
+  readonly verificationGatewayAddress!: string;
+  wallet!: BlsWalletWrapper;
+  _index: number;
+  _address: string;
+
+  constructor(
+    constructorGuard: any,
```

Comment:
Prefer better typing over `any`.
```suggestion
    constructorGuard: Record<string, unknown>,
```